core/src/mindustry/world/blocks/production/Drill.java

```diff
@@ -66,6 +67,19 @@ public Drill(String name){
         idleSoundVolume = 0.003f;
     }
 
+    @Override
+    public void drawRequestConfigTop(BuildRequest req, Eachable<BuildRequest> list){
+        Tile tile = req.tile();
+        if(tile == null) return;
```

Comment:
Technical fix: You should also skip drawing if `!req.worldContext` (so schematics don't grab world state).